Sources/BraintreeVenmo/BTVenmoClient.swift

```diff
@@ -0,0 +1,357 @@
+            appSwitchCompletion(nil, returnURL.error)
+            return
+            
+        case .canceled:
+            apiClient.sendAnalyticsEvent("ios.pay-with-venmo.appswitch.handle.cancel")
+            appSwitchCompletion(nil, nil)
+            return
+            
+        default:
+            // should not happen
+            break
+        }
+    }
+
+    func performAppSwitch(with appSwitchURL: URL, shouldVault vault: Bool, completion: @escaping (BTVenmoAccountNonce?, Error?) -> Void) {
+        if let _ = application as? UIApplication {
+            UIApplication.shared.open(appSwitchURL) { success in
+                self.invokedOpenURLSuccessfully(success, shouldVault: vault, completion: completion)
+            }
+        } else {
+            application.open(appSwitchURL) { success in
+                self.invokedOpenURLSuccessfully(success, shouldVault: vault, completion: completion)
+            }
+        }
+    }
+
+    func invokedOpenURLSuccessfully(_ success: Bool, shouldVault vault: Bool, completion: @escaping (BTVenmoAccountNonce?, Error?) -> Void) {
+        shouldVault = success && vault
+
+        if success {
+            apiClient.sendAnalyticsEvent("ios.pay-with-venmo.appswitch.initiate.success")
+            BTVenmoClient.venmoClient = self
+            self.appSwitchCompletion = completion
+        } else {
+            apiClient.sendAnalyticsEvent("ios.pay-with-venmo.appswitch.initiate.error.failure")
+            completion(nil, BTVenmoError.appSwitchFailed)
+        }
+    }
+
+    // MARK: - Vaulting Methods
+
+    func vault(_ nonce: String) {
+        let venmoAccount: [String: String] = ["nonce": nonce]
+        let parameters: [String: Any] = ["venmoAccount": venmoAccount]
+
+        apiClient.post("v1/payment_methods/venmo_accounts", parameters: parameters) { body, _, error in
+            if let error = error as? NSError {
+                if error.code == BTCoreConstants.networkConnectionLostCode {
+                    self.apiClient.sendAnalyticsEvent("ios.pay-with-venmo.network-connection.failure")
+                }
+
+                self.apiClient.sendAnalyticsEvent("ios.pay-with-venmo.vault.failure")
+                self.appSwitchCompletion(nil, error)
+                return
+            }
+
+            guard let body else {
+                self.appSwitchCompletion(nil, BTVenmoError.invalidBodyReturned)
+                return
+            }
+
+            let venmoAccountJSON: BTJSON = body["venmoAccounts"][0]
+            let venmoAccountNonce: BTVenmoAccountNonce = BTVenmoAccountNonce.venmoAccount(with: venmoAccountJSON)
+
+            self.apiClient.sendAnalyticsEvent("ios.pay-with-venmo.vault.success")
+            self.appSwitchCompletion(venmoAccountNonce, venmoAccountJSON.asError())
+            return
+        }
+    }
+
+    // MARK: - App Switch Methods
+
+    func verifyAppSwitch(with configuration: BTConfiguration, error: inout Error?) -> Bool {
```

Comment:
🧶 : is the `inout` still necessary? A `try / catch` could be more expressive.

Reply:
Done! https://github.com/braintree/braintree_ios/pull/970/commits/8e8bc6fcf0a99a23f7bb5a18e78ba84217ae7d9b